src/shared/sol-iio.h

```diff
@@ -0,0 +1,194 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief Routines to access Linux Industrial I/O (iio) devices under Solleta
+ */
+
+struct sol_iio_device;
+struct sol_iio_channel;
+struct sol_iio_buffer;
+
+/**
+ * Callback to read data from buffer
+ */
+typedef void (*sol_iio_reader_cb)(struct sol_iio_device *device, struct sol_iio_buffer *buffer, void *data);
+
+/**
+ * Open an IIO device
+ *
+ * @param device_id Id of iio device. It's the number which identifies device
+ * on file system. Can be found at '/sys/bus/iio/devices/iio:deviceX'
+ * @param buffer_size size of reading buffer. If 0, will use device default;
+ * if -1 will disable buffer - and all reads will be performed on channel files
+ * on sysfs.
+ * @param trigger_name Name of IIO trigger that will be used on this device. If NULL
+ * or empty, will try to use device current trigger. If device has no current trigger,
+ * will create a 'sysfs_trigger' and use it.
+ *
+ * @return A new IIO handle
+ */
+struct sol_iio_device *sol_iio_open(int device_id, int buffer_size, const char *trigger_name);
```

Comment:
similar to what I said in `sol_iio_add_channel()` we better have a `struct sol_iio_config` with all the configuration at once instead of multiple calls to setup buffer, callback, etc.